viewing_party/party.py

```diff
@@ -1,23 +1,178 @@
-# ------------- WAVE 1 --------------------
-
+# 1st function in Wave 1
 def create_movie(title, genre, rating):
-    pass
 
-# -----------------------------------------
-# ------------- WAVE 2 --------------------
-# -----------------------------------------
+    if title is None or genre is None or rating is None:
+        return None
+
+    new_movie = {"title": title, "genre": genre, "rating": rating}
+
+    if isinstance(title, str) or isinstance(genre, str) or isinstance(rating, int):
+        return {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+    else:
+        return None
+
+
+# 2nd function in Wave 1
+def add_to_watched(user_data, movie):
+
+    (user_data["watched"]).append(movie)
+    return user_data
+
+
+# 3rd function in Wave 1
+def add_to_watchlist(user_data, movie):
+
+    watchlist = user_data.get("watchlist", [])
+
+    watchlist.append(movie)
+    user_data["watchlist"] = watchlist
+    
+    return user_data
+
+
+# 4th function in Wave 1
+def watch_movie(user_data, title):
+
+    move_movie = False
+    index = 0
+    movie = None
+
+    for index in range(len(user_data["watchlist"])):
+        if title == user_data["watchlist"][index]["title"]:
+            move_movie = True
+            movie_index = index
+            movie = user_data["watchlist"][index]
+            print(user_data["watchlist"][index])
+    if move_movie:
+        (user_data["watched"]).append(movie)
+        del user_data["watchlist"][movie_index]
+    return user_data
+
+
+# 1st function in Wave 2
+def get_watched_avg_rating(user_data):
+
+    movie_count = 0
+    ratings_sum = 0
+    ratings_avg = 0
+
+    if user_data["watched"]:
+
+        for index in range(len(user_data["watched"])):
+
+            movie_count += 1
+            ratings_sum += user_data["watched"][index]["rating"]
+        ratings_avg = ratings_sum/movie_count
+    return ratings_avg
+
+
+# 2nd function in Wave 2
+def get_most_watched_genre(user_data):
+
+    most_watched_genre = user_data.get("watched",[])
+    
+    if not most_watched_genre: 
+        return None
+
+    genres = {}
+    for movie in most_watched_genre:
+        genre = movie.get("genre")
+        if genre in genres:
+            genres[genre] += 1
+        else:
+            genres[genre] = 1
+
+    most_watched_genre = max(genres, key=genres.get)

```

Comment:
Here, you're re-assigning the value of `most_watched_genre` to value of `max(...)`. This variable feels very separate from when it was created at the beginning of the function.

I recommend using two different variables. Perhaps the first variable can be called `watched_movies`? If we use that name, we can now see the separation between `most_watched_genre`:

```python
def get_most_watched_genre(user_data):
    watched_movies = user_data.get("watched",[])
    
    if not watched_movies: 
        return None

    genres = {}
    for movie in watched_movies:
        genre = movie.get("genre")
        if genre in genres:
            genres[genre] += 1
        else:
            genres[genre] = 1

    most_watched_genre = max(genres, key=genres.get)

    return most_watched_genre 
```